handlers/auth_handler.go

```diff
@@ -472,3 +472,237 @@ func LogoutHandler(c echo.Context) error {
+// @Failure      400 {object} echo.HTTPError  "Bad request"
+// @Failure      404 {object} echo.HTTPError  "User not found"
+// @Failure      429 {object} echo.HTTPError  "Too many requests"
+// @Failure      500 {object} echo.HTTPError  "Internal server error"
+// @Router       /v1/auth/forgot-password [post]
+func ForgotPasswordHandler(c echo.Context) error {
+	logger := c.Logger()
+
+	var req ForgotPasswordRequest
+	if err := c.Bind(&req); err != nil {
+		logger.Error("Invalid forgot password request payload:", err)
+		return echo.ErrBadRequest
+	}
+
+	if req.Email == "" {
+		logger.Error("Email is required.")
+		return &echo.HTTPError{
+			Code:    http.StatusBadRequest,
+			Message: "email field is required",
+		}
+	}
+
+	newCrypto := crypto.NewCrypto()
+	user := models.User{}
+
+	emailPseudo, err := newCrypto.HashData([]byte(req.Email), "HMAC-SHA-256")
+	if err != nil {
+		logger.Errorf("Failed to hash email: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	if err := db.Conn.Where("email_pseudonym = ?", emailPseudo).First(&user).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			logger.Error("User not found for password reset.")
+			return c.JSON(http.StatusOK, GenericResponse{
+				Message: "If an account with that email exists, we've sent a password reset link",
+			})
+		}
+
+		logger.Errorf("Failed to find user: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	recentReset := models.PasswordReset{}
+	if err := db.Conn.Where("user_id = ? AND created_at > ?", user.ID, time.Now().Add(-5*time.Minute)).
+		First(&recentReset).Error; err == nil {
+		logger.Info("Recent password reset email already sent")
+		return &echo.HTTPError{
+			Code:    http.StatusTooManyRequests,
+			Message: "Please wait 5 minutes before requesting another password reset email",
+		}
+	}
+
+	token, err := crypto.GenerateRandomString("prt_", 32, "hex")
+	if err != nil {
+		logger.Errorf("Failed to generate password reset token: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	expiresAt := time.Now().Add(24 * time.Hour)
+
+	passwordReset := models.PasswordReset{}
+	if err := db.Conn.Where("user_id = ? AND is_used = ?", user.ID, false).
+		Assign(models.PasswordReset{
+			UserID:    user.ID,
+			Token:     token,
+			ExpiresAt: expiresAt,
+			IsUsed:    false,
+		}).FirstOrCreate(&passwordReset).Error; err != nil {
+		logger.Errorf("Failed to check existing password reset tokens: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	emailBytes, err := newCrypto.DecryptData(user.EmailEncrypted, "AES-GCM")
+	if err != nil {
+		logger.Errorf("Failed to decrypt user email: %v", err)
+		return echo.ErrInternalServerError
+	}
+	email := string(emailBytes)
+	fullName := ""
+
+	baseUrl := commons.GetEnv("BASE_URL", "https://api.queuedroid.com")
+	resetLink := commons.GetEnv("PASSWORD_RESET_URL", "https://queuedroid.com") + "/reset-password?token=" + token
```

Comment:
We should use `EMAIL_VERIFICATION_URL` variable to avoid introducing new variables.